pkg/termination/termination_test.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package termination
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tektoncd/pipeline/pkg/logging"
+)
+
+func TestExistingFile(t *testing.T) {
+	tmpFile, err := ioutil.TempFile(os.TempDir(), "tempFile")
+	if err != nil {
+		log.Fatal("Cannot create temporary file", err)
+	}
+	// Remember to clean up the file afterwards
+	defer os.Remove(tmpFile.Name())
+
+	logger, _ := logging.NewLogger("", "test termination")
+	defer logger.Sync()
+	output := []v1alpha1.PipelineResourceResult{}
+	now := time.Now()
+	output = append(output, v1alpha1.PipelineResourceResult{
+		Key:   "StartedAt",
+		Value: now.Format(time.RFC3339),
+	})
+
+	WriteMessage(logger, tmpFile.Name(), output)
+
+	output = []v1alpha1.PipelineResourceResult{}
+	now = time.Now()
+	output = append(output, v1alpha1.PipelineResourceResult{
+		Key:   "FinishedAt",
+		Value: now.Format(time.RFC3339),
+	})
+
+	WriteMessage(logger, tmpFile.Name(), output)
```

Comment:
Perhaps `WriteMessage` should return an error, that we can `t.Fatal` on if we see here. `logger.Fatalf` inside of `WriteMessage` will make test failures harder to find, and it's probably just better hygiene anyway.

This will also let us remove `logger` as an arg to `WriteMessage`, since it was only taking it to be able to call `Fatalf` on it in error scenarios.

Reply:
Done with latest commit